crates/exex/src/exex.rs

```diff
@@ -70,116 +68,53 @@ impl<Node: FullNodeComponents> KakarotRollup<Node> {
-                    })
-                    .collect::<eyre::Result<Vec<_>>>()
-                    .map(|txs| (block, txs, receipts))
-            })
-            .collect::<eyre::Result<Vec<_>>>()?;
-
-        // Process each block and its transactions.
-        for (block, txs, _) in blocks {
-            // Execute the block and handle the result.
-            match execute_block(&mut self.db, block, txs.clone()).await {
-                Ok((block, bundle, _, _)) => {
-                    // Seal the block.
-                    let block = block.seal_slow();
-                    // After the block is executed, we want to run the get_env program
-                    //
-                    // The goal is to store in a Cairo program, via hint, all the information about
-                    // the block
-                    self.execute_get_env(block.block.clone(), txs, chain.tip().number)?;
-                    // Insert the block into the database.
-                    self.db.insert_block_with_bundle(&block, bundle)?;
-                    info!(
-                        block_hash = %block.hash(),
-                        transactions = block.body.len(),
-                        "Block submitted, executed and inserted into database"
-                    );
-                }
-                Err(err) => {
-                    // Log an error if the block execution fails.
-                    error!(
-                        %err,
-                        block_hash = %block.hash(),
-                        transactions = block.body.len(),
-                        "Failed to execute block"
-                    );
-                }
+                // Retrieve the output of the program
+                let mut output_buffer = String::new();
+                res.vm.write_output(&mut output_buffer).unwrap();
+                println!("Program output: \n{}", output_buffer);
+
+                // Extract the execution trace
+                let trace = res.relocated_trace.clone().unwrap_or_default();
+
+                // Extract the relocated memory
+                let memory = res
+                    .relocated_memory
+                    .clone()
+                    .into_iter()
+                    .map(|x| x.unwrap_or_default())
+                    .collect();
+
+                // Extract the public and private inputs
+                //
+                // We want to store the public input in the database in order to use them to run
+                // the prover
+                let public_input = res.get_air_public_input()?;
+                let private_input = res.get_air_private_input();
+
+                // Commit the execution trace to the database
+                self.commit_cairo_execution_traces(
+                    committed_chain.tip().number,
+                    trace,
+                    memory,
+                    public_input,
+                    private_input,
+                )?;
             }
```

Comment:
This function was very important, allowing to re-execute our blocks via revm as long as we do not have the run via our own evm. It allows us to run the process as a whole: ExEx + re-execution + hint in any program. It seems early to remove it, right? For me it takes away a test environment and I think we could keep it for now.

Reply:
I may have misunderstood some part, is it required to re-run with revm to have the initial partial state?

Reply:
No, not really, in fact the process so far was as follows:
- We receive a block with transactions via the ExEx that we mock for our tests (that is to say that we voluntarily inject a real block from the mainnet with a real transaction in it).
- We take this block and we re-execute here the transactions that are inside via the revm (which will be done in the future by our OS that will just have to be injected here when it is ready), which allows us to do all kinds of tests or executions in parallel.

I know that one day we will have to remove this function because our goal is obviously not to reuse revm but it seems a bit early to remove it, wdyt?